commit 29f7b376d399acf6c1e54267e5214aca3549a001
Author: Ross Lagerwall <ross.lagerwall@citrix.com>
Date:   Fri Feb 1 14:42:28 2019 +0000

    Revert "scsi: libfc: Add WARN_ON() when deleting rports"
    
    [ Upstream commit d8f6382a7d026989029e2e50c515df954488459b ]
    
    This reverts commit bbc0f8bd88abefb0f27998f40a073634a3a2db89.
    
    It added a warning whose intent was to check whether the rport was still
    linked into the peer list. It doesn't work as intended and gives false
    positive warnings for two reasons:
    
    1) If the rport is never linked into the peer list it will not be
    considered empty since the list_head is never initialized.
    
    2) If the rport is deleted from the peer list using list_del_rcu(), then
    the list_head is in an undefined state and it is not considered empty.
    
    Signed-off-by: Ross Lagerwall <ross.lagerwall@citrix.com>
    Reviewed-by: Hannes Reinecke <hare@suse.com>
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/scsi/libfc/fc_rport.c b/drivers/scsi/libfc/fc_rport.c
index 372387a450df..1797e47fab38 100644
--- a/drivers/scsi/libfc/fc_rport.c
+++ b/drivers/scsi/libfc/fc_rport.c
@@ -184,7 +184,6 @@ void fc_rport_destroy(struct kref *kref)
 	struct fc_rport_priv *rdata;
 
 	rdata = container_of(kref, struct fc_rport_priv, kref);
-	WARN_ON(!list_empty(&rdata->peers));
 	kfree_rcu(rdata, rcu);
 }
 EXPORT_SYMBOL(fc_rport_destroy);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1110_29f7b376d399acf6c1e54267e5214aca3549a001.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1110_29f7b376d399acf6c1e54267e5214aca3549a001.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
